### PR TITLE
Illustrate setting up an infra repository

### PR DESCRIPTION
**Pull request by tthyer**
This is one way to set up an infra repo Sage-style.

This is being placed in -Workflows as I'm not sure where it should live -- maybe in IT? But if so I'd like to have people who are interested in its existence to have sufficient permissions on the repo.

One of these repos will typically go hand in hand with the creation of an AWS account or set of accounts. Something's needed to set up the CI user, anyway -- it cannot be set up from this repo. This account is currently not associated with any AWS accounts so it's not actually fully set up. I'm thinking to leave it that way and disable the workflow, with instructions for what they need to do to do a real setup.

### CODE REVIEW

**Review by zaro0508 — COMMENTED** [review]
hmm, i'm conflicted about this GH template.  It seems like the most appropriate place for this is Sage-Bionetworks-IT org?  However i'm not sure whether working from this is more helpful than pointing users to at an existing, up to date, working repo with CI setup.  Also i'm not sure IT wants to maintain this because it can become out of date rather quickly which would become a maintenance burden.  In fact, there is already a [Sage-Bionetworks-IT/aws-account-template](https://github.com/Sage-Bionetworks-IT/aws-account-template) repo for the same purpose that is completely out of date 😞

**Comment by tthyer** [comment]
@zaro0508 I'm happy to move this repository to -IT iff the likely contributors can have admin on the repository. That you don't want to support it, however, is a good reason for it not to be in -IT, right? I think ideally we would have have something like a Sage-Bionetworks-IBC organization that this would live in.

I do know about Sage-Bionetworks-IT/aws-account-template. However, it is archived, and as you say, terribly out of date.

I think it's definitely worth discussing whether it is preferable to have a template repository or documentation which references the most ideal example. @thomasyu888, @BrunoGrandePhD, any thoughts on this?  I personally prefer the template repository because we can quickly scaffold a new project with it. However, Khai's right, this is only useful if we keep it up to date.

**Comment by BrunoGrandePhD** [comment]
I agree that there's the risk that this won't be maintained. That said, if that risk exists for both documentation and a template repository, I prefer investing into the more practical resource, _i.e._ the template repository. Also, given that we're moving towards creating more project-specific AWS accounts, this repository will probably see more usage than [Sage-Bionetworks-IT/aws-account-template](https://github.com/Sage-Bionetworks-IT/aws-account-template) and thus is more likely to be maintained as a group effort among folks who create and manage these infra repos. 

As for the location, I suggest moving this repository to the Sage-Bionetworks organization because that will ensure that everyone at Sage has access to the repository and thus has the ability to use the template. We can also create a GitHub team of infra folks with write-access and included as code owners for PRs. I'm basing this suggestion on the following quote ([source](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository)) and the fact that membership in the Sage-Bionetworks-IT organization is intentionally limited. (Note that the fact that the template is open-source isn't enough; otherwise, our template dropdown menus would be way longer.)

> After you make your repository a template, anyone with access to the repository can generate a new repository with the same directory structure and files as your default branch.

**Comment by thomasyu888** [comment]
I agree with @BrunoGrandePhD and @tthyer .  

I think it makes sense for this to be part of the Sage-Bionetworks organization.  I also think that if we include this in the instructions in how to create AWS resources via cloudformation, we will almost with 100% certainty have to update the template which I think is better than not having any template for people to work from.  IT doesn't necessarily have to maintain this template repo, but I know IT does review a the infrastructure repositories, and from those reviews, we can propagate any relevant changes to this repo if necessary.

**Comment by tthyer** [comment]
Based on previous discussion, this template is going to stay in Sage-Bionetworks-Workflows. We won't be moving it to a Github organization where there's no admin member from our team, since we are the people who value having this template and will be responsible for maintaining it.

This PR has no approvals. Does anyone have feedback, of whatever magnitude, to give on this repository? Please speak now.

File: Pipfile
Comment (BrunoGrandePhD) [inline Pipfile:7]: Do we want to avoid the `networkx` issues? If so, we could pin this to 3.8.

File: README.md
Comment (BrunoGrandePhD) [inline README.md:12]: This is redundant with earlier steps, no?

**Review by BrunoGrandePhD — APPROVED** [review]
A few nits, but otherwise, it looks good to me! 🚀 

As for the location, I don't know if you missed my [suggestion](https://github.com/Sage-Bionetworks-Workflows/infra-template/pull/1#issuecomment-924202979) to move this repository to `Sage-Bionetworks`, where we can have admin access (not `Sage-Bionetworks-IT`): 

> As for the location, I suggest moving this repository to the Sage-Bionetworks organization because that will ensure that everyone at Sage has access to the repository and thus has the ability to use the template. We can also create a GitHub team of infra folks with write-access and included as code owners for PRs. I'm basing this suggestion on the following quote ([source](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository)) and the fact that membership in the Sage-Bionetworks-IT organization is intentionally limited. (Note that the fact that the template is open-source isn't enough; otherwise, our template dropdown menus would be way longer.)
> 
> > After you make your repository a template, anyone with access to the repository can generate a new repository with the same directory structure and files as your default branch.

File: Pipfile
Comment (tthyer) [reply Pipfile:7]: I'm sort of ok with this except that I have not found it's always true that there are networkx issues if you use 3.9. Can you give more details re when that problem arises?

File: Pipfile
Comment (tthyer) [reply Pipfile:11]: No, not really. Let's say you have a lambda repository: there are going to be things in it that should be legitimately in `[packages]` b/c the lambda depends on them.

On the other hand an argument can be made that this repository should be title `sceptre-infra-repository`, then all these things could just go in packages. I have placed them in `[dev-packages]` because @zaro0508 previously expressed strong feelings about this.

**Comment by tthyer** [comment]
@BrunoGrandePhD I liked your suggestion about moving the repository but there are some issues with it too. I have been talking to our other members about it offline. I'll reach out to you offline as well to discuss what we're waiting on there.

**Pull request merged**
